lib/elixir/lib/registry.ex

```diff
@@ -1318,6 +1318,47 @@ defmodule Registry do
     end
   end
 
+  @doc """
+  Works like `select/2`, but only returns the number of matching records.
+
+  ## Examples
+
+  In the example below we register the current process under different
+  keys in a unique registry but with the same value:
+
+      iex> Registry.start_link(keys: :unique, name: Registry.CountSelectTest)
+      iex> {:ok, _} = Registry.register(Registry.CountSelectTest, "hello", :value)
+      iex> {:ok, _} = Registry.register(Registry.CountSelectTest, "world", :value)
+      iex> Registry.count_select(Registry.CountSelectTest, [{{:_, :_, :value}, [], [true]}])
+      2
+  """
+  @spec count_select(registry, spec) :: non_neg_integer()
+  def count_select(registry, spec)
+      when is_atom(registry) and is_list(spec) do
+    spec =
+      for part <- spec do
+        case part do
+          {{key, pid, value}, guards, select} ->
+            {{key, {pid, value}}, guards, select}
+
+          _ ->
+            raise ArgumentError,
+                  "invalid match specification in Registry.select_count/2: #{inspect(spec)}"
+        end
+      end
+
+    case key_info!(registry) do
+      {_kind, partitions, nil} ->
+        Enum.reduce(0..(partitions - 1), 0, fn partition_index, acc ->
+          count = :ets.select_count(key_ets!(registry, partition_index), spec)
+          acc + count
+        end)
+
+      {_kind, 1, key_ets} ->
+        :ets.select_count(key_ets, spec)
+    end
```

Comment:
We have this exact logic in `count_match` too. In fact, the one in count_match is a bit more optimized, so please extract the one in count_match and reuse it. :)

Reply:
Thank you for the guidance, José!

If I understand correctly, for `count_match` when `key_info!` returns `{:unique, partitions, nil}` - the logic is slightly different. There we can rely on given `key` to find the `key_ets`. While for `count_select` that's not the case.. we have to traverse all partitions to find the total count. Am I missing something?

Here https://github.com/elixir-lang/elixir/pull/11405/commits/a35e45e4e87e9833ef1bea227ce87610a1d9014a I extracted the logic out, in order to reuse it, but since that one particular case is different - I pass the logic in anonymous function, so that `count_select` and `count_match` can define their own way to handle the case. 
What would be the better way to deal with this situation?

_P.S. I couldn't come up with better naming for private functions, went with more descriptive for now_

Reply:
Yeah, sorry I missed this nuance. I think we should revert this one. :)

Reply:
Done! =)